PyPDF2/generic.py

```diff
@@ -578,6 +578,29 @@ def writeToStream(self, stream, encryption_key):
 
     @staticmethod
     def readFromStream(stream, pdf):
+        def getNextObjPos(p, p1, remGens, pdf):
+            l = pdf.xref[remGens[0]]
+            for o in l:
+                if p1 > l[o] and p < l[o]:
+                    p1 = l[o]
+            if len(remGens) == 1:
+                return p1
+            else:
+                return getNextObjPos(p, p1, remGens[1:], pdf)
+
+        def readUnsizedFromSteam(stream, pdf):
```

Comment:
I don't understand this function. How does it know where the ream ends? Couldn't there be multiple objects in a stream?

Can you point me to a resource where I can read up on PDF streams in general? I would appreciate that a lot :pray:

Reply:
From my analysis/tests, the stream were most of the time compressed/encoded 
Also we have to remember that this function will  only be used to load some data when the PDF file is corrupted.

I peek up some information in the standard
https://opensource.adobe.com/dc-acrobat-sdk-docs/standards/pdfstandards/pdf/PDF32000_2008.pdf
and also in those:
https://www.adobe.com/technology/pdfs/presentations/KingPDFTutorial.pdf
https://www.oreilly.com/library/view/developing-with-pdf/9781449327903/ch01.html

for ref also so test files
https://github.com/pdf-association/pdf20examples